crates/sdk/src/tx.rs

```diff
@@ -2940,6 +2921,41 @@ async fn used_asset_types<P, R, K, N>(
     Ok(asset_types)
 }
 
+/// Constructs the batched tx from the provided list. Returns also the data for
+/// signing
+pub fn build_batch(
+    mut txs: Vec<(Tx, SigningTxData)>,
+) -> Result<(Tx, Vec<SigningTxData>)> {
+    let (mut batched_tx, sig_data) = txs.pop().ok_or_else(|| {
```

Comment:
As per the previous comment, we want to execute transactions in the batch in the same order of the insertion.
Can we take the first one here?

Reply:
This should be fixed in 47ccfd35d8ed43a23bf2c66a85df34f1c8af4112